### PR TITLE
Fixup response code on incorrect credentials

### PR DESCRIPTION
**Pull request by vishesh92**
### Description

This PR fixes https://github.com/apache/cloudstack/issues/8662
The issue got introduced in 4.19. When we try to login, we check login credentials using different types of authenticators. While verifying in case of wrong password with `oauth2` authenticator and there are no providers setup for it, it throws a CloudRuntime Exception. Because the error isn't caught, we respond with 200 status code but with empty response. This results in UI getting stuck. This PR fixes this by skipping the authentication check if the secret code is null.

While logging in, if the user doesn't exist it fails with a valid error
image

If the user exists, but the password is wrong then loading starts on top and the UI gets stuck. I had to clean up my local storage to make the browser work again.
image

In order to reproduce, use username as "admin" or any other user which exists. For password, enter anything except the password.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by vishesh92** [comment]
@blueorangutan package

File: server/src/main/java/com/cloud/user/AccountManagerImpl.java [hunk 1]
```diff
@@ -2650,7 +2650,8 @@ private UserAccount getUserAccount(String username, String password, Long domain
                     continue;
                 }
             }
-            if (secretCode != null && !authenticator.getName().equals("oauth2")) {
+            if ((secretCode != null && !authenticator.getName().equals("oauth2"))
+                    || (secretCode == null && authenticator.getName().equals("oauth2"))) {
```
Comment (sureshanaparti) [inline server/src/main/java/com/cloud/user/AccountManagerImpl.java:2654]: can replace string "_oauth2_" with relevant constant?

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by vishesh92** [comment]
@blueorangutan package

**Comment by vishesh92** [comment]
@blueorangutan test

**Review by harikrishna-patnala — APPROVED** [review]
code LGTM. @vishesh92 can you please quickly verify the same when oauth is enabled.

**Comment by vishesh92** [comment]
> code LGTM. @vishesh92 can you please quickly verify the same when oauth is enabled.

Same issue was happening even if oauth was enabled because the provider name is not there in the request params. It's resolved after this change.

**Comment by harikrishna-patnala** [comment]
> > code LGTM. @vishesh92 can you please quickly verify the same when oauth is enabled.
> 
> Same issue was happening even if oauth was enabled because the provider name is not there in the request params. It's resolved after this change.

thanks @vishesh92

**Review by sureshanaparti — APPROVED** [review]
code lgtm

**Comment by vishesh92** [comment]
@blueorangutan package

**Comment by vishesh92** [comment]
@blueorangutan package

**Comment by vishesh92** [comment]
@blueorangutan package

**Comment by vishesh92** [comment]
@blueorangutan package

**Comment by vishesh92** [comment]
@blueorangutan package

**Comment by vishesh92** [comment]
@blueorangutan package

**Comment by vishesh92** [comment]
@blueorangutan package

**Comment by vishesh92** [comment]
@blueorangutan package

**Comment by vishesh92** [comment]
@blueorangutan package

**Comment by vishesh92** [comment]
@blueorangutan package

**Comment by vishesh92** [comment]
@blueorangutan test

**Comment by vishesh92** [comment]
@blueorangutan package

**Review by DaanHoogland — APPROVED** [review]
clgtm tested in lab env; login as admin with a wrong password.
before the page needed refreshing, now the error is shown and a new attempt can be made.

**Pull request merged**